File: artifacts/tulipindicators/decompose/src/add.c
int ti_add_start(const double *options)
{
  (void) options;
  return 0;
}


/*** DEPENDENCIES:
***/


int ti_add(int size, const double * const *inputs, const double *options, double * const *outputs)
{
  const double *in1 = inputs[0];
  const double *in2 = inputs[1];
  (void) options;
  double *output = outputs[0];
  int i;
  for (i = 0; i < size; i += 1)
  {
    output[i] = in1[i] + in2[i];
  }

  return 0;
}


/*** DEPENDENCIES:
***/

File: artifacts/tulipindicators/translate_without_break_down/src/add.rs
use crate::*;

pub fn ti_add_start(options: Option<&[f64]>) -> i32 {
    let _ = options; // Explicitly ignore the parameter
    0
}
pub fn ti_add(
    size: i32,
    inputs: Option<&[&[f64]]>,
    options: Option<&[f64]>,
    outputs: Option<&mut [&mut [f64]]>,
) -> i32 {
    // Check for None in inputs and outputs (equivalent to NULL checks in C)
    if inputs.is_none() || outputs.is_none() {
        return 0;
    }

    let inputs = inputs.unwrap();
    let outputs = outputs.unwrap();

    // Ensure we have at least two input slices and one output slice
    if inputs.len() < 2 || outputs.is_empty() {
        return 0;
    }

    let in1 = inputs[0];
    let in2 = inputs[1];
    let output = &mut outputs[0];

    // Ensure all slices have at least `size` elements
    if in1.len() < size as usize || in2.len() < size as usize || output.len() < size as usize {
        return 0;
    }

    // Perform the addition operation
    for i in 0..size as usize {
        output[i] = in1[i] + in2[i];
    }

    0
}
